packages/core/src/domain/internalMonitoring/startMonitoringBatch.ts

```diff
@@ -1,11 +1,15 @@
 import { Batch, HttpRequest } from '../../transport'
-import type { Configuration, EndpointBuilder, MonitoringMessage } from '../..'
+import type { Configuration, EndpointBuilder, Context } from '../..'
 
-export function startMonitoringBatch(configuration: Configuration) {
-  const primaryBatch = createMonitoringBatch(configuration.internalMonitoringEndpointBuilder!)
+export function startMonitoringBatch<T extends Context>(
```

Comment:
Indeed, `startLoggerBatch` and `startMonitoringBatch` were almost the same thing 🤔 
I had in mind to only keep this abstraction while having both systems but we could use that for logs as well and keep it.

For RUM, it seems a bit trickier since there are more behaviors related to the batch (upsert, unload, replica app id).

Wdyt of only mutualizing that with logs?

Reply:
Indeed, for RUM it is trickier. I would still factorize it though:
* while only used by RUM, `upsert` and `unload` are implemented in `Batch`, so it wouldn't hurt to expose it through the more abstract `startBatch` function.
* for replica specificities, we could have a `replicaContext`

It could be done in a future task though